.github/workflows/build-all.yaml

```diff
@@ -0,0 +1,50 @@
+name: BuildAll
+
+on:
+  workflow_call:
```

Comment:
I think we shouldn't switch the pr / push main jobs to this. But instead change the on to `workflow_dispatch` so we can trigger it manually when needed.

https://docs.github.com/en/webhooks/webhook-events-and-payloads#workflow_dispatch